aeron-cluster/src/main/java/io/aeron/cluster/Election.java

```diff
@@ -666,6 +708,49 @@ private int followerReplay(final long nowNs)
         return workCount;
     }
 
+    private int followerLogReplication(final long nowNs)
+    {
+        int workCount = 0;
+
+        if (null == logReplication)
+        {
+            if (appendPosition < logReplicationPosition)
+            {
+                logReplication = consensusModuleAgent.newLogReplication(
+                    leaderMember.archiveEndpoint(), leaderRecordingId, logReplicationPosition);
+                workCount++;
+            }
+            else
+            {
+                state(FOLLOWER_REPLAY, nowNs);
+            }
+        }
+        else
+        {
+            workCount += logReplication.doWork();
+            if (logReplication.isDone())
+            {
+                // Which position should be updated here???
+                appendPosition = logReplication.replicatedLogPosition();
+                assert appendPosition == logReplicationPosition;
```

Comment:
We don't use `assert` in Java code. Either verify the position in business logic or not.